src/validation.cpp

```diff
@@ -1851,6 +1851,18 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         flags |= SCRIPT_VERIFY_NULLDUMMY;
     }
 
+    // SEGWIT2X signalling.
+    if ( VersionBitsState(pindex->pprev, chainparams.GetConsensus(), Consensus::DEPLOYMENT_SEGWIT2X, versionbitscache) == THRESHOLD_ACTIVE &&
```

Comment:
@Saicere Yes, good point, miners could manually flip on bit 1 as soon as they see bit 4 locked in.  Just figured making that automatic might get BIP141 locked in one difficulty adjustment sooner.   As I understand the current code we could be looking at 6 weeks from bit 4 reaching 80% to segwit activation, which seems unnecessary.